arcgis-ios-sdk-samples/Display information/Add graphics with renderer/AddGraphicsWithRendererViewController.swift

```diff
@@ -0,0 +1,171 @@
+        let heartGeometry = makeHeartGeometry(center: origin, sideLength: 10e5)
+        let heartGraphic = AGSGraphic(geometry: heartGeometry, symbol: nil)
+        curveGraphicsOverlay.graphics.add(heartGraphic)
+        
+        // Add the overlays to the map view.
+        mapView.graphicsOverlays.addObjects(from: [pointGraphicsOverlay, lineGraphicsOverlay, polygonGraphicsOverlay, curveGraphicsOverlay])
+    }
+    
+    func makePointGraphicsOverlay() -> AGSGraphicsOverlay {
+        // Create a simple marker symbol.
+        let pointSymbol = AGSSimpleMarkerSymbol(style: .diamond, color: .green, size: 10)
+        // Create a graphics overlay for the points.
+        let pointGraphicsOverlay = AGSGraphicsOverlay()
+        // Create and assign a simple renderer to the graphics overlay.
+        pointGraphicsOverlay.renderer = AGSSimpleRenderer(symbol: pointSymbol)
+        return pointGraphicsOverlay
+    }
+    
+    func makeLineGraphicsOverlay() -> AGSGraphicsOverlay {
+        // Create a simple line symbol.
+        let lineSymbol = AGSSimpleLineSymbol(style: .solid, color: .blue, width: 5)
+        // Create a graphics overlay for the polylines.
+        let lineGraphicsOverlay = AGSGraphicsOverlay()
+        // Create and assign a simple renderer to the graphics overlay.
+        lineGraphicsOverlay.renderer = AGSSimpleRenderer(symbol: lineSymbol)
+        return lineGraphicsOverlay
+    }
+    
+    func makeSquarePolygonGraphicsOverlay() -> AGSGraphicsOverlay {
+        // Create a simple fill symbol.
+        let squareSymbol = AGSSimpleFillSymbol(style: .solid, color: .yellow, outline: nil)
+        // Create a graphics overlay for the square polygons.
+        let squareGraphicsOverlay = AGSGraphicsOverlay()
+        // Create and assign a simple renderer to the graphics overlay.
+        squareGraphicsOverlay.renderer = AGSSimpleRenderer(symbol: squareSymbol)
+        return squareGraphicsOverlay
+    }
+    
+    func makeCurvedPolygonGraphicsOverlay() -> AGSGraphicsOverlay {
+        // Create a simple fill symbol with outline.
+        let curvedLineSymbol = AGSSimpleLineSymbol(style: .solid, color: .black, width: 1)
+        let curvedFillSymbol = AGSSimpleFillSymbol(style: .solid, color: .red, outline: curvedLineSymbol)
+        // Create a graphics overlay for the polygons with curve segments.
+        let curvedGraphicsOverlay = AGSGraphicsOverlay()
+        // Create and assign a simple renderer to the graphics overlay.
+        curvedGraphicsOverlay.renderer = AGSSimpleRenderer(symbol: curvedFillSymbol)
+        return curvedGraphicsOverlay
+    }
+    
+    /// Create a heart-shape geometry with Bezier and elliptic arc segments.
+    ///
+    /// - Parameters:
+    ///   - center: The center of the square that contains the heart shape.
+    ///   - sideLength: The side length of the square.
+    /// - Returns: A heart-shape geometry.
+    func makeHeartGeometry(center: AGSPoint, sideLength: Double) -> AGSGeometry? {
```

Comment:
`AGSPoint`'s initializer takes `Double` for coordinates, thus I used double for the side length, instead of the conventional `CGFloat` (for core graphics classes such as `CGRect` and `CGSize`) to avoid casting back to double.